src/NuGet.Core/NuGet.Common/PathUtil/FileUtility.cs

```diff
@@ -92,7 +92,12 @@ public static void Replace(Action<string> writeSourceFile, string destFilePath)
                 throw new ArgumentNullException(nameof(destFilePath));
             }
 
-            var tempPath = GetTempFilePath(Path.GetDirectoryName(destFilePath));
+            string? destFileDirectory = Path.GetDirectoryName(destFilePath);
+            if (destFileDirectory is null)
+            {
+                throw new ArgumentException(paramName: destFilePath, message: "Path.GetDirectoryName(destFilePath) returned null");
```

Comment:
>Path.GetDirectoryName(destFilePath) returned null [](http://example.com/codeflow?start=79&length=49)

This probably needs to be localized.

Reply:
I don't expect any customers to ever see this message, so no, I don't think it needs to be localized. It's only feasible if someone using `NuGet.Common` (part of the Client SDK) as part of their app calls the API with bad input (or if someone modifying NuGet starts calling this in a way that is buggy). Frankly, I think it's a bad API, NuGet shouldn't be in the business of proving and supporting trivial file utility APIs.

I also think it's a bad idea to parse exception messages. .NET Core changed some exception messages from what they were on the .NET Framework (I think ArgumentNullException's message is an example), and exception messages that are localized will have unexpected results when running on machines not using the same language as the developer who wrote code trying to parse the exception message. Even for tests this is a bad idea, since I don't want to exclude people running their devbox in something other than `en-US` from being able to run tests locally.

Reply:
I'm okay with the answer. Suggestion for message: "Not a valid file path".